core/src/Streamly/Internal/Data/ParserK/Type.hs

```diff
@@ -748,21 +763,23 @@ toParser parser = ParserD.Parser step initial extract
     step cont a = do
         r <- cont (Chunk a)
         return $ case r of
-            Done n b -> assert (n <= 1) (ParserD.Done (1 - n) b)
+            Done n b -> assert (n <= 1) (ParserD.SDone n b)
             Error _ e -> ParserD.Error e
-            Partial n cont1 -> assert (n <= 1) (ParserD.Partial (1 - n) cont1)
-            Continue n cont1 -> assert (n <= 1) (ParserD.Continue (1 - n) cont1)
+            Partial n cont1 -> assert (n <= 1) (ParserD.SPartial n cont1)
+            Continue n cont1 -> assert (n <= 1) (ParserD.SContinue n cont1)
 
     extract cont = do
         r <- cont None
         case r of
             -- This is extract so no input has been given, therefore, the
-            -- translation here is (0 - n) rather than (1 - n).
-            Done n b ->  assert (n <= 0) (return $ ParserD.Done (negate n) b)
+            -- translation here is (1 + n) rather than n.
+            -- XXX How should we visualize (1 + n)?
+            -- XXX Go forward by 1 element on extract?
```

Comment:
Go back by n but consume the current element?
Is this correct? Should it be `n - 1` here, not `1 + n`?
How are the tests passing? Check if we test this.

Reply:
Even though I put `n - 1` here, or `n + 1` here, the test cases are passing. We need to write a more elaborate test case for `fromParserK`

Reply:
Ah, we have some fusion `RULES` due to which the test case isn't run like it should.